junit-platform-suite-engine/src/main/java/org/junit/platform/suite/engine/SuiteTestDescriptor.java

```diff
@@ -0,0 +1,153 @@
+import org.junit.platform.launcher.TestPlan;
+import org.junit.platform.launcher.core.LauncherConfig;
+import org.junit.platform.launcher.core.LauncherFactory;
+import org.junit.platform.suite.api.SuiteDisplayName;
+
+/**
+ * {@link TestDescriptor} for tests based on the JUnit Platform Launcher API.
+ *
+ * <h3>Default Display Names</h3>
+ *
+ * <p>The default display name is the simple name of the class.
+ *
+ * @since 1.8
+ * @see SuiteDisplayName
+ */
+final class SuiteTestDescriptor extends AbstractTestDescriptor {
+
+	static final String SEGMENT_TYPE = "suite";
+
+	private final SuiteLauncherDiscoveryRequestBuilder requestBuilder;
+
+	private TestPlan testPlan;
+	private Launcher launcher;
+
+	SuiteTestDescriptor(UniqueId id, Class<?> suiteClass, SuiteConfiguration configuration) {
+		super(id, getSuiteDisplayName(suiteClass), ClassSource.from(suiteClass));
+		// @formatter:off
+		UniqueId requestingSuiteId = requireNoCycles(configuration).parentSuiteId()
+				.map(parentId -> UniqueIdHelper.append(parentId, getUniqueId()))
+				.orElseGet(this::getUniqueId);
+		// @formatter:on
+		SuiteLauncherDiscoveryRequestBuilder requestBuilder = new SuiteLauncherDiscoveryRequestBuilder();
+		this.requestBuilder = requestBuilder.configureRequestingSuiteId(requestingSuiteId);
+	}
+
+	private SuiteConfiguration requireNoCycles(SuiteConfiguration configuration) {
+		configuration.parentSuiteId().ifPresent(parentSuiteId -> {
+			UniqueId fullSuiteId = UniqueIdHelper.append(parentSuiteId, getUniqueId());
+			Supplier<String> message = () -> String.format(
+				"Configuration error: The suite configuration may not contain a cycle [%s]", fullSuiteId);
+			Preconditions.condition(!UniqueIdHelper.containCycle(fullSuiteId, SEGMENT_TYPE), message);
+		});
+		return configuration;
+	}
+
+	UniqueId uniqueIdInSuite(TestIdentifier testDescriptor) {
+		Preconditions.notNull(testDescriptor, "uniqueId most not be null");
+		UniqueId uniqueIdInTestPlan = UniqueId.parse(testDescriptor.getUniqueId());
+		UniqueId uniqueIdInSuite = getUniqueId();
+		return UniqueIdHelper.append(uniqueIdInSuite, uniqueIdInTestPlan);
+	}
+
+	SuiteTestDescriptor addDiscoveryRequestFrom(Class<?> testClass) {
+		Preconditions.condition(testPlan == null, "discovery request can not be modified after discovery");
+		requestBuilder.addRequestFrom(testClass);
+		return this;
+	}
+
+	SuiteTestDescriptor addDiscoveryRequestFrom(UniqueId uniqueId) {
+		Preconditions.condition(testPlan == null, "discovery request can not be modified after discovery");
+		requestBuilder.addRequestFrom(uniqueId);
+		return this;
+	}
+
+	void discover() {
+		Preconditions.condition(testPlan == null, "discovery can only happen once");
+
+		LauncherDiscoveryRequest request = requestBuilder.build();
+		// @formatter:off
+		LauncherConfig launcherConfig = LauncherConfig.builder()
+				.enableTestExecutionListenerAutoRegistration(false)
+				.enablePostDiscoveryFilterAutoRegistration(false)
```

Comment:
Consider explicitly disabling the discovery of launcher-discovery listeners via SPI. https://github.com/junit-team/junit5/pull/2457

Reply:
We can keep this enabled. While we try to hide that a suite is discovering and executing tests using the junit platform internally, we don't need to hide that launcher discovery is started.